hackmcgill/dashboard
# build: Upgrading dependencies for compatibility with Node 20 and general maintenance

### List of changes:

- Upgrading all dependencies to latest stable versions (where possible)
- Several API adjustments required, but functionality is retained (as far as I was able to test)
- Dependency override added for react-table in order to be compatible with React 18. Seems that this dependency has changed quite a bit so we are forced remain at the older version and override the React dependency
- styled-components is fixed to v5 since v6 typing is broken currently :(
- formik is fixed at 2.1.0 as newer versions interfere with weird reset form setup we have for multi-page forms (ex., Hacker application)
- Various unused dependencies removed
- Did a lint of the files (seems its been a while), so a lot of the changes are just that

### Type of change:

Please delete options that aren't relevant.

- [x] Breaking change (fix or feature that ~would~ _may_ cause existing functionality to not work as expected)

### How did you do this?
Basically just using `ncu -u` (npm-check-upgrade) to update all dependencies and then updating system Node version and then doing some testing/tinkering. Read up on the various package APIs for those that were broken.

### How to test:
Each working component needs to be tested to ensure no functionality has been broken. Unfortunately, my back end doesn't seem to be set up completely? Or perhaps it is just meant to be like that, so I am not able to test completely. But, I have test what I can and everything seems to be great!

### Questions:

### PR Checklist:


### Screenshots:
